cli/args/flags.rs

```diff
@@ -971,6 +1031,60 @@ pub fn flags_from_vec(args: Vec<String>) -> clap::error::Result<Flags> {
   Ok(flags)
 }
 
+/// The fast flag parser for the most common options.
+fn fast_flags_from_vec(args: &[String], env_vars: &EnvVars) -> Option<Flags> {
+  // We don't parse environment variables in this path, just abort if we see any
+  // that impact parsing.
+  if !env_vars.is_empty() {
```

Comment:
@lucacasonato Added a bailout if we see env vars -- also these env vars are parsed in a single, consistent place .